tests/Kernel/Patterns/ComponentsPageTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\Kernel;
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\ui_patterns_library\Controller\PatternsLibraryController;
+
+/**
+ * Class ComponentsPageTest.
+ *
+ * Purpose of this test class is to make sure
+ * that all patterns will work without fatal errors with minimal dependencies.
+ */
+class ComponentsPageTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'system',
+    'ui_patterns',
+    'ui_patterns_library',
+    'oe_theme_helper',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->installConfig(['system']);
+    $this->container->get('theme_installer')->install(['oe_theme']);
+    $this->container->get('theme_handler')->setDefault('oe_theme');
+    $this->container->set('theme.registry', NULL);
+  }
+
+  /**
+   * Test a components page rendering.
+   *
+   * @throws \Exception
+   */
+  public function testComponentsPageRendering(): void {
+    $controller = new PatternsLibraryController($this->container->get('plugin.manager.ui_patterns'));
+    $renderable_array = $controller->overview();
+    $this->container->get('renderer')->renderRoot($renderable_array);
```

Comment:
Do you think it would be possible to assert that the page has been rendered without issue ?
It would be nicer to have a more self-explanatory test because at first sight, it seems that this method (_which is supposed to have assertions_) doesn't test anything.

Reply:
Yes. Your doubts is correct. I have found that this approach didn't covered all cases. I reworked testing approach just by changing \Drupal\Tests\oe_theme\Kernel\RenderingTest class.